File: C++/Hash/완주하지_못한_선수.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    int i;

    sort(participant.begin(), participant.end());
    sort(completion.begin(), completion.end());

    for (i = 0; i < completion.size(); i++) {
        if (participant[i] != completion[i]) break;
    }
    return participant[i];
}

File: Python/Hash/완주하지 못한 선수.py
def solution(participant, completion):
    participant.sort()
    completion.sort()

    for p, c in zip(participant, completion):
        if p != c:
            return p
    
    return participant.pop()

p = ["leo", "kiki", "eden"]
c = ["eden", "kiki"]

print(solution(p, c))